### PR TITLE
Fix `mypy` errors in `erosion_operation.py`

### PR DESCRIPTION
**Pull request by tianyizheng02**
### Describe your change:

- Fixed type hints in `erosion_operation.py` to resolve some `mypy` errors (see #8070)
- Did some minor renaming/refactoring

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Python files are placed inside an existing directory.
* [ ] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by tianyizheng02** [comment]
@cclauss Mind taking a look? It's pretty much the same changes from #8595

File: digital_image_processing/morphological_operations/erosion_operation.py [hunk 1]
```diff
@@ -23,7 +24,8 @@ def rgb_to_gray(rgb: np.ndarray) -> np.ndarray:
 def gray_to_binary(gray: np.ndarray) -> np.ndarray:
     """
     Return binary image from gray image
-    >>> gray_to_binary(np.array([[127, 255, 0]]))
+
+    >>> gray2binary(np.array([[127, 255, 0]]))
```
Comment (cclauss) [inline digital_image_processing/morphological_operations/erosion_operation.py]: The function name on line 24 is `gray_to_binary()`, not `gray2binary()`.

**Comment by tianyizheng02** [comment]
@cclauss Pinging for review, the typo that you had brought up in your last review had already been fixed

**Review by cclauss — APPROVED** [review]
Nice!

**Pull request merged**